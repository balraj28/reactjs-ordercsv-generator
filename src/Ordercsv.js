import React from 'react';
import { usePapaParse, useCSVDownloader } from 'react-papaparse';

function Ordercsv(props) {
  const { jsonToCSV } = usePapaParse();
  const { CSVDownloader, Type, } = useCSVDownloader();
  return (
    <>
      {/* CSV BLOCK START */}
      {props.filename && props?.data?.length > 0 && (
        <>
          <h2 className="fs-title" data-testid="filename">{`${props.filename}.csv`}</h2>
          <textarea data-testid="file1-csv-data" value={jsonToCSV(props?.data, { header: false })} readOnly />
          <CSVDownloader
            type={Type.Button}
            config={{ header: false }}
            filename={`${props.filename}`}
            data={props?.data}
          >Download</CSVDownloader>
        </>
      )}
      {/* CSV BLOCK END */}
    </>
  );
}

export default Ordercsv;
