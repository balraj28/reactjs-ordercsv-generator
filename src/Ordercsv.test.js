import { render, screen} from '@testing-library/react';
import Ordercsv from './Ordercsv';
import '@testing-library/jest-dom';

test('should render the csv block with valid CSV data snapshot', () => {
  const csvdata = [
    { prod: 'paint', Avg: '1.0' },
    { prod: 'laptop', Avg: '3.0' }
  ];
  render(<Ordercsv data={csvdata} filename={'order_log'} />);
  expect(screen.getByTestId('file1-csv-data')).toMatchSnapshot();
});

test('should not render the csv block without valid filename', () => {
  const csvdata = [{ prod: 'paint', Avg: '1.0' },
  { prod: 'laptop', Avg: '3.0' }];
  render(<Ordercsv data={csvdata} filename={''} />);
  expect(screen.queryByText(/_0.csv/)).toBe(null);
});

test('should not render the csv block without valid data', () => {
  render(<Ordercsv data={[]} filename={'order_log'} />);
  expect(screen.queryByText(/order_log_0.csv/)).toBe(null);
});

test('should render the download option without valid data', async () => {
  render(<Ordercsv data={[]} filename={''} />);
  expect(screen.queryByText(/Download/)).toBe(null);
});

test('should render the download option with valid data', async () => {
  render(<Ordercsv data={[{ prod: 'paint', Avg: '1.0' }]} filename={'order_log'} />);
  expect(screen.queryByText(/Download/)).toBeInTheDocument();
});
