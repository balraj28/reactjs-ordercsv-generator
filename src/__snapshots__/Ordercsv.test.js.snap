// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the csv block with valid CSV data 1`] = `
<textarea
  data-testid="file1-csv-data"
  readonly=""
>
  paint,1.0
laptop,3.0
</textarea>
`;
